org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/compiler/regression/PrimitiveInPatternsTestSH.java

```diff
@@ -2302,6 +2302,28 @@ int foo4(int myInt) {
 				""");
 	}
 
+	public void testIncompatiblePrimitiveInInstanceof() {
+		runNegativeTest(new String[] {
+				"X.java",
+				"""
+				public class X  {
+					void foo() {
+						if (this instanceof int i)
+							return;
+					}
+				}
+				"""
+				},
+				"""
+				----------
+				1. ERROR in X.java (at line 3)
+					if (this instanceof int i)
+					    ^^^^^^^^^^^^^^^^^^^^^
+				Incompatible conditional operand types X and int
+				----------
+				""");
+	}
```

Comment:
I so much like javac's message : `incompatible types: X cannot be converted to int` - I don't know where this conditional operand types terminology originated but I am OK with that staying and perhaps being taken up separately.

Reply:
'conditional operand types' has been in use for instanceof expressions for at least very very long time (like: for ever?), see, e.g., org.eclipse.jdt.core.tests.compiler.regression.CastTest.test027()

I filed #3119 for follow-up.